plugins/editor/class.editor.plugin.php

```diff
@@ -613,7 +616,38 @@ public function base_render_before($Sender) {
         $c->addDefinition('editorFileInputName', $this->editorFileInputName);
         $Sender->setData('_editorFileInputName', $this->editorFileInputName);
         // Save allowed file types
-        $c->addDefinition('allowedFileExtensions', json_encode(c('Garden.Upload.AllowedFileExtensions')));
+        $allowedFileExtensions = c('Garden.Upload.AllowedFileExtensions');
+        $imageExtensions = ['gif', 'png', 'jpeg', 'jpg', 'bmp', 'tif', 'tiff', 'svg'];
+
+        $allowedImageExtensions = array_intersect($allowedFileExtensions, $imageExtensions);
+
+        $c->addDefinition('allowedImageExtensions', json_encode($allowedImageExtensions));
+        $c->addDefinition('allowedFileExtensions', json_encode($allowedFileExtensions));
+
+        $allowedMimeTypes = [];
+        foreach($allowedFileExtensions as $ext) {
+            if ($mime = $this->lookupMime($ext)) {
+                $allowedMimeTypes = array_merge($allowedMimeTypes, $mime);
+            }
+        }
+
+        $allowedImageMimeTypes = [];
+        foreach($allowedImageExtensions as $ext) {
+            if ($mime = $this->lookupMime($ext)) {
+                $allowedImageMimeTypes = array_merge($allowedImageMimeTypes, $mime);
+            }
+        }
+
+        // prepend extensions with a '.'
+        $allowedFileExtensions = array_map(function($str) { return '.'.$str; }, $allowedFileExtensions);
+        $accept = implode(',', array_merge($allowedFileExtensions, $allowedMimeTypes));
+        $Sender->setData('Accept', $accept);
+
+        // prepend extensions with a '.'
+        $allowedImageExtensions = array_map(function($str) { return '.'.$str; }, $allowedImageExtensions);
```

Comment:
You're also using the same anonymous function code in two different areas.  You can save it to a variable and pass that to `array_map` for each call.  Also, the function code should probably be broken up across multiple lines.